src/components/exercises/ExercisesList.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC, useState, useEffect } from 'react';
+
+import { Button } from 'antd';
+
+import { Card, Loading } from '../custom';
+
+import * as api from '../../api';
+
+import { Exercise } from '../../types';
+
+import ExerciseItem from './ExerciseItem';
+
+import './ExercisesList.css';
+
+const ExercisesList: FC<{
+    history: any
+}> = ({ history }) => {
+  // recup exercises
+	const [ exercises, setExercises ] = useState<Array<Exercise>>([]);
+
+	useEffect(() => {
+		api.Exercises.all()
+			.then((exercises: Exercise[]) =>
+				setExercises(exercises)
+			);
+	}, []);
+
+	const handleDelete = (exerciseId: number) => () => {
+		api.Exercises.delete(exerciseId);
+		setExercises(
```

Comment:
What happens if the delete fails here? A client is always a mirror for a server, so I would argue that you should ask the server for the updated list of exercices. TBD

Reply:
What about using promises ? We could do like this :
`api.Exercises.delete((exerciseId))`
    `.then(() => {`
      `setExercises(exercises.filter(exercise => exercise.id !== exerciseId))`
    `})`
    `.catch(() => {`
      `alert("La suppression a échoué");`
    `});`